File: src/main/java/org/danwatt/voronipostals/VoroniController.java
package org.danwatt.voronipostals;

import org.rapidoid.annotation.Controller;
import org.rapidoid.annotation.GET;
import org.rapidoid.annotation.Page;

import java.util.Arrays;
import java.util.List;

@Controller
public class VoroniController {

    @Page("/map")
    public Object map() {
        return "";
    }

    @GET("/nearby/counties/{point}")
    public List<PostalCode> nearby(String point) {
        String[] parts = point.split(",");
        double lat = Double.parseDouble(parts[0]);
        double lon = Double.parseDouble(parts[1]);
        return PostalSource.getInstance().getNearbyCounties(lat,lon);
    }

    @GET("/nearby/postals/{point}")
    public List<PostalCode> nearbyPostals(String point) {
        String[] parts = point.split(",");
        double lat = Double.parseDouble(parts[0]);
        double lon = Double.parseDouble(parts[1]);
        return PostalSource.getInstance().getPostalsAroundPoint(lat,lon);
    }

    @GET("/postals/union/{postals}")
    public List<PostalCode> getUnion(String postals) {
        return PostalSource.getInstance().getUnion(Arrays.asList(postals.split(",")));
    }
}

File: src/main/java/org/danwatt/voronipostals/VoroniController.kt
package org.danwatt.voronipostals

import org.rapidoid.annotation.Controller
import org.rapidoid.annotation.GET
import org.rapidoid.annotation.Page
import java.lang.Double.parseDouble
import java.util.Collections.singletonList

@Controller
class VoroniController(private val instance: PostalSource = PostalSource.instance) {

    @Page("/map")
    fun map(): Any {
        return ""
    }

    @GET("/nearby/counties/{point}")
    fun nearby(point: String): List<GeoContainer> {
        val parts = point.split(",".toRegex()).dropLastWhile(String::isEmpty)
        val lat = parseDouble(parts[0])
        val lon = parseDouble(parts[1])
        return PostalQueries.collectNearbyCounties(lat, lon, instance.countyIndex)
    }

    @GET("/nearby/postals/{point}")
    fun nearbyPostals(point: String): List<GeoContainer> {
        val parts = point.split(",".toRegex()).dropLastWhile(String::isEmpty)
        val lat = parseDouble(parts[0])
        val lon = parseDouble(parts[1])

        return PostalQueries.collectNearbyPostals(lat, lon, instance.postalIndex, instance.postalCodes)
    }

    @GET("/postals/union/{postals}")
    fun getUnion(postals: String): List<GeoContainer> {
        val splitPostals = postals.split(",".toRegex()).dropLastWhile(String::isEmpty)
        val union = PostalQueries.unionPostalCodes(splitPostals, instance.postalCodes) ?: return emptyList()
        return singletonList(union)

    }


}
